site/app/controllers/PollController.php

```diff
@@ -0,0 +1,236 @@
+        parent::__construct($core);
+    }
+
+    /**
+    * @Route("/{_semester}/{_course}/polls", methods={"GET"})
+    * @return MultiResponse
+    */
+    public function showPollsPage() {
+        if ($this->core->getUser()->accessAdmin()) {
+            return MultiResponse::webOnlyResponse(
+                new WebResponse(
+                    'Poll',
+                    'showPollsInstructor',
+                    $this->core->getQueries()->getTodaysPolls(),
+                    $this->core->getQueries()->getOlderPolls(),
+                    $this->core->getQueries()->getFuturePolls()
+                )
+            );
+        }
+        else {
+            return MultiResponse::webOnlyResponse(
+                new WebResponse(
+                    'Poll',
+                    'showPollsStudent',
+                    $this->core->getQueries()->getTodaysPolls(),
+                    $this->core->getQueries()->getOlderPolls()
+                )
+            );
+        }
+    }
+
+    /**
+    * @Route("/{_semester}/{_course}/polls/viewPoll", methods={"POST"})
+    * @return MultiResponse
+    */
+    public function showPoll() {
+        $poll = $this->core->getQueries()->getPoll($_POST["poll_id"]);
+        if ($this->core->getUser()->accessAdmin()) {
+            return MultiResponse::webOnlyResponse(
+                new WebResponse(
+                    'Poll',
+                    'showPollInstructor',
+                    $poll
+                )
+            );
+        }
+        else {
+            return MultiResponse::webOnlyResponse(
+                new WebResponse(
+                    'Poll',
+                    'showPollStudent',
+                    $poll
+                )
+            );
+        }
+    }
+
+    /**
+    * @Route("/{_semester}/{_course}/polls/newPoll", methods={"GET"})
+    * @AccessControl(role="INSTRUCTOR")
+    * @return MultiResponse
+    */
+    public function showNewPollPage() {
+        return MultiResponse::webOnlyResponse(
+            new WebResponse(
+                'Poll',
+                'showNewPollPage'
+            )
+        );
+    }
+
+
+    /**
+    * @Route("/{_semester}/{_course}/polls/newPoll", methods={"POST"})
+    * @AccessControl(role="INSTRUCTOR")
+    * @return MultiResponse
+    */
+    public function addNewPoll() {
+        $response_count = $_POST["response_count"];
+        $responses = array();
+        $answers = array();
+        $orders = array();
+        for ($i = 0; $i < $response_count; $i++) {
+            $responses[$_POST["option_id_" . $i]] = $_POST["response_" . $i];
+            $orders[$_POST["option_id_" . $i]] = $_POST["order_" . $i];
+            if (isset($_POST["is_correct_" . $i]) && $_POST["is_correct_" . $i] == "on") {
+                $answers[] = $_POST["option_id_" . $i];
+            }
+        }
+        $this->core->getQueries()->addNewPoll($_POST["name"], $_POST["question"], $responses, $answers, $_POST["release_date"], $orders);
```

Comment:
You should add some error checking in case there is no `name`, `question`, `release_date`